### ISSUE TITLE
Problem with Docker

### ISSUE THREAD

**Issue by AnastasiaKremenetskaya** [HansSchouten/Laravel-Pagebuilder#89]
![Снимок экрана 2021-02-20 в 21 22 17](https://user-images.githubusercontent.com/46082850/108604899-b6577780-73c1-11eb-80bd-d14761f6e80d.png)

Get this error though using in DB_HOST my container name (in .env)

My .env
```
DB_CONNECTION=mysql
DB_HOST=rdn_db_1
DB_PORT=33060
DB_DATABASE=db
DB_USERNAME=root
DB_PASSWORD=secret
```

My config/pagebuilder
```
    'storage' => [
        'use_database' => true,
        'database' => [
            'driver'    => 'mysql',
            'host'      => env('DB_HOST').':'.env('DB_PORT',3306),
            'database'  => env('DB_DATABASE'),
            'username'  => env('DB_USERNAME'),
            'password'  => env('DB_PASSWORD'),
            'charset'   => 'utf8',
            'collation' => 'utf8_unicode_ci',
            'prefix'    => 'pagebuilder__',
        ],
        'uploads_folder' => storage_path('app/pagebuilder/uploads')
    ],

```

If I set `DB_HOST=host.docker.internal` (can't use it in production) there's no database error, but router not working as it should work. And shows Asset not found to all assets in package.

![Снимок экрана 2021-02-20 в 21 29 30](https://user-images.githubusercontent.com/46082850/108605047-ba37c980-73c2-11eb-9852-a68ee5fd5a2c.png)

Above problems appear only when using PHPageBuilder with docker.

**Comment by dsultanr** [comment]
![image](https://user-images.githubusercontent.com/1683476/108634698-d3f11380-748b-11eb-8b6c-ab0950681070.png)

why do you have different ports in .env and config?

I think DB issue could not be the cause of "asset not found" error. What is the full path of app.css in browser?

**Comment by AnastasiaKremenetskaya** [comment]
I don't have different ports. `env('DB_PORT',3306), ` means that if DB_PORT in env is empty then use default 3306 port.
Full path is http://localhost/assets/websitemanager/app.css

**Comment by dsultanr** [comment]
but you have 33060 port in your .env

**Comment by AnastasiaKremenetskaya** [comment]
Yes, that's why default 3306 port won't be used. 3306 is used only when DB_PORT in .env is empty, my DB_PORT is not empty.

**Comment by ehsantalkhouncheh** [comment]
Hi @AnastasiaKremenetskaya , I have same issue did you find a solution for that?

**Comment by AnastasiaKremenetskaya** [comment]
@ehsantalkhouncheh No, forced to discard usage

**Comment by bluekable** [comment]
Not sure if this applies to everyone using Docker but it definitely applies when installing Laravel for dev using Laravel Sail.

In `vendor/hansschouten/phpagebuilder/src/Core/helpers.php` in the phpb_current_full_url() method the full URL is generated using `SERVER['SERVER_NAME']` which in the case of these docker containers in windows and WSL2 (not sure about in mac OS) is generally 0.0.0.0.

Then in the phpb_current_relative_url() method the relative URL is generated using both the returned value from the above mentioned phpb_current_full_url() and the value in `phpb_config('general.base_url');` which in most cases is probably `http://localhost`.

Changing the ENV or config value for base_url doesn't solve the issue as redirects are sent to 0.0.0.0 then which breaks in the WSL2/Docker environment.

**WARNING!:** below is a quick fix for getting up and running in dev environments and all changes will be overwritten if you update your laravel-pagebuilder package. 

A workaround I have found for now is to modify the `phpb_current_full_url()` to just return `'http://localhost' . urldecode($_SERVER['REQUEST_URI']);`.
